### DISCUSSION TITLE
App over bluetooth on id4++

### DISCUSSION THREAD

**Discussion by SaxMachine** [BimmerGestalt/AAIdrive#585] [Q&A]
Hello, thanks for this fantastic app. 
I have a quick question!
It is possibile enabling the apps over bluetooth on id4++?
If not, could it be possible an firmwares upgrading to id5 for enabling this function? 
Thank you!

**Comment by hufman** [comment]
As far as I know, there is no possibility to enable Bluetooth Apps on ID4, and no firmware update to convert ID4 hardware to ID5: You'd have to swap out the entire infotainment headunit to achieve this.

**Reply by SaxMachine** [reply]
Thanks I have read many forums that upgrade id4 hardware to id5 and also id6 (the latter with many bugs but is working.) 
I'd be curious to try.

**Reply by Vasmares** [reply]
my mini has ID5 (2020).
Where did you read it was upgradeable to ID 6 ?
I would love to try it out

**Reply by fookyuanloh** [reply]
ID5 to Id6 (lite) can be done with just coding
No need to reflash etc.
Was told that it is more of a UI skin change 

Do-able with the common coding tools

**Reply by Vasmares** [reply]
Thanks

**Reply by hufman** [reply]
I'm curious on what Mini ID5 vs ID6 looks like, share pictures if you try it!
I suspect the latest Mini generation, with the homescreen weather widget, is ID6, but that's my only guess.

**Comment by TBGR** [comment]
Some iDrive4s can be flashed to 5 but it brings issues witb it. Check your map version, if it is NEXT you can't do it. If it is EVO you can. But your iDrive still has half the RAM of an actual iDrive 5 unit. After the update using the navigation on really long routes with music playing from a source different from radio might cause the iDrive to freeze and reboot itself once every hour or so.

**Reply by SaxMachine** [reply]
Yes my HU is upgradable and I have the Evo version. The first version of id5 firmware is configured on the same hw of ID4, 2 GB of ram. 
I'm go to try to install the 2016/17 version of id5 because is the best release for my HU.

**Reply by fookyuanloh** [reply]
> Yes my HU is upgradable and I have the Evo version. The first version of id5 firmware is configured on the same hw of ID4, 2 GB of ram. I'm go to try to install the 2016/17 version of id5 because is the best release for my HU.

@SaxMachine 
What is the outcome?
Upgraded the ID4 to ID5, and able to connect to AAI via BT?

**Reply by SaxMachine** [reply]
Yes! It is working very fine! ;)

**Comment by fookyuanloh** [comment]
Mind to share how did you get it to work?
Did you remove and re-add vehicle on MYBMW app prior to BT connection?

**Reply by SaxMachine** [reply]
But you have updated you id4++ ti id5? By the way you have to remove all associations smartphone and idrive. Remove cache and data from AAidrive apps, remove the app, reinstall all and try again.

**Comment by fookyuanloh** [comment]
Id4 to ID5+ (with Id6 skin)
How do you remove cache?

Uninstall then reinstall back MyBMW + AAI
I did remove my car from MYBMW app before, but I didn't uninstall the app
I just re-add vehicle

Perhaps I re-try, do everything as fresh setup
